server/planning/commands/delete_spiked_items.py

```diff
@@ -30,123 +39,125 @@ class DeleteSpikedItems(Command):
-            return
+    expiry_datetime = now - timedelta(minutes=expire_interval)
 
-        expiry_datetime = now - timedelta(minutes=expire_interval)
+    try:
+        await delete_spiked_events(expiry_datetime)
+    except Exception as e:
+        logger.exception(e)
 
-        try:
-            self._delete_spiked_events(expiry_datetime)
-        except Exception as e:
-            logger.exception(e)
+    try:
+        await delete_spiked_planning(expiry_datetime)
+    except Exception as e:
+        logger.exception(e)
 
-        try:
-            self._delete_spiked_planning(expiry_datetime)
-        except Exception as e:
-            logger.exception(e)
+    unlock(lock_name)
 
-        unlock(lock_name)
+    logger.info(f"{log_msg} Completed deleting spiked items.")
+    remove_locks()
 
-        logger.info("{} Completed deleting spiked items.".format(self.log_msg))
-        remove_locks()
 
-    def _delete_spiked_events(self, expiry_datetime):
-        logger.info("{} Starting to delete spiked events".format(self.log_msg))
-        events_service = get_resource_service("events")
+async def delete_spiked_events(expiry_datetime):
+    log_msg = log_msg_context.get()
+    logger.info(f"{log_msg} Starting to delete spiked events")
+    events_service = EventsAsyncService()
 
-        events_deleted = set()
-        series_to_delete = dict()
+    events_deleted = set()
+    series_to_delete = dict()
 
-        # Obtain the full list of Events that we're to process first
-        # As subsequent queries will change the list of returned items
-        events = dict()
-        for items in events_service.get_expired_items(expiry_datetime, spiked_events_only=True):
-            events.update({item[ID_FIELD]: item for item in items})
+    # Obtain the full list of Events that we're to process first
+    # As subsequent queries will change the list of returned items
+    events = dict()
+    async for items in events_service.get_expired_items(expiry_datetime, spiked_events_only=True):
+        events.update({item[ID_FIELD]: item for item in items})
 
-        for event_id, event in events.items():
-            if event.get("recurrence_id") and event["recurrence_id"] not in series_to_delete:
-                spiked, events = self.is_series_expired_and_spiked(event, expiry_datetime)
-                if spiked:
-                    series_to_delete[event["recurrence_id"]] = events
-            else:
-                events_service.delete_action(lookup={"_id": event_id})
-                events_deleted.add(event_id)
+    for event_id, event in events.items():
+        if event.get("recurrence_id") and event["recurrence_id"] not in series_to_delete:
+            spiked, events = await is_series_expired_and_spiked(event, expiry_datetime)
+            if spiked:
+                series_to_delete[event["recurrence_id"]] = events
+        else:
+            await events_service.delete_action(lookup={"_id": event_id})
```

Comment:
There is no `delete_action` in the new resource services, instead there is a `delete_many` method